source/s3_meta_request.c

```diff
@@ -159,6 +167,10 @@ int aws_s3_meta_request_init_base(
         vtable->sign_request = s_s3_meta_request_sign_request_default;
     }
 
+    if (vtable->init_signing_date_time == NULL) {
```

Comment:
Normally we have vtables as const.  Is it going to be a problem that this isn't?

Reply:
Ah, I was just trying to find a C way of providing a default implementation.  It doesn't mutate after initialization, so I happy to make it const and then cast it away during setup if that's more correct.

Reply:
Oh, right.  I was thinking about that incorrectly.  I'll fix that.